### PR TITLE
LANraragi: Update to API 0.7.2

### LINKED ISSUE #4278: [Bug] LANraragi: Broken
## Device information
- Tachiyomi version: 0.10.4
- Android version: 8.0.0

## Source information
- Name: LANraragi
- Extension version: 1.2.1

## Steps to reproduce
1. Point at server hostname. Give API key.
2. Open source.

### Expected behavior
I see my library.

### Actual behavior
Screenshot_20200831-165244

### PR DESCRIPTION
**Pull request by RePod**
Uses new search, extract, and thumbnail endpoints.
extract endpoint now uses POST instead of GET.
If an API key is provided it is base64 encoded and added to the headers.

Fixes #4278

### CODE REVIEW

File: src/all/lanraragi/src/eu/kanade/tachiyomi/extension/all/lanraragi/LANraragi.kt [hunk 1]
```diff
@@ -49,20 +52,23 @@ open class LANraragi : ConfigurableSource, HttpSource() {
...
-        val uri = getApiUriBuilder("/api/extract")
-        uri.appendQueryParameter("id", id)
+        val uri = getApiUriBuilder("/api/archives/$id/extract")
 
         return listOf(
             SChapter.create().apply {
                 val uriBuild = uri.build()
 
-                url = "${uriBuild.encodedPath}?${uriBuild.encodedQuery}"
+                url = "${uriBuild.encodedPath}"
```
Comment (arkon) [inline src/all/lanraragi/src/eu/kanade/tachiyomi/extension/all/lanraragi/LANraragi.kt:61]: Can remove the string interpolation here.

Comment (RePod) [reply src/all/lanraragi/src/eu/kanade/tachiyomi/extension/all/lanraragi/LANraragi.kt:61]: Would `uriBuild.encodedPath.toString()` suffice?

Comment (arkon) [reply src/all/lanraragi/src/eu/kanade/tachiyomi/extension/all/lanraragi/LANraragi.kt:61]: It should, if it doesn't already give you a string.

**Comment by arkon** [comment]
Remember to bump the version number in the `build.gradle` file as well.

**Pull request merged**